src/Span.h

```diff
@@ -1375,4 +1431,206 @@ void Span::serialize_json(RapidJsonWriter &writer) const
     writer.EndObject();
 }
 
-} // namespace zipkin
\ No newline at end of file
+namespace __impl
+{
+
+struct Span2 {
+    enum Kind {
+        UNKNOWN,
```

Comment:
we leave null for unknown vs an unknown constant. as long as you handle the same semantics up to you though